lib/api/objectDelete.js

```diff
@@ -22,10 +22,6 @@ const versionIdUtils = versioning.VersionID;
  */
 function objectDelete(authInfo, request, log, cb) {
     log.debug('processing request', { method: 'objectDelete' });
-    if (authInfo.isRequesterPublicUser()) {
```

Comment:
I am still not sure that object delete can be performed by non-owners, but based on the behavior we saw, this should be ok.

Reply:
Will undo this change and adjust the tests accordingly